### PR TITLE
feat: add funtionality for multiple minters

### PR DESCRIPTION
**Pull request by whylee259**
- ItemNFT
    - multiple minter: add burnApproval and management feature for it
    - mint, mintBatch => safeMint, safeMintBatch(like openzeppelin wizard does)
       -  https://docs.openzeppelin.com/contracts/4.x/wizard
- MultiToken
    - multiple minter

### CODE REVIEW

**Review by egonspace — APPROVED** [review]
LGTM

File: contracts/token/ERC1155/MultiToken.sol [hunk 1]
```diff
@@ -89,22 +99,27 @@ contract MultiToken is
...
         onlyOwner
         whenBurnableEnabled
         whenNotPaused
     {
         burnApprovals[burner] = approved;
-        emit BurnPermissionApproval(burner, approved);
+        emit BurnApproval(burner, approved);
+    }
+
+    function setMintApproval(address minter, bool approved) external onlyOwner {
```
Comment (brew0722) [inline contracts/token/ERC1155/MultiToken.sol]: It looks like missing the `whenNotPaused` compared `setBurnApproval`

Comment (whylee259) [reply contracts/token/ERC1155/MultiToken.sol]: Fixed. thanks!

**Review by brew0722 — APPROVED** [review]
LGTM

**Pull request merged**